test/variants/ub_eq08.v3

```diff
@@ -0,0 +1,22 @@
+//@execute 0=true;1=false;2=true;3=false;4=true;5=false
```

Comment:
Can you write a few more test cases, some smaller and some larger, and a couple that are polymorphic?